api/tests/unit/features/versioning/test_unit_versioning_tasks.py

```diff
@@ -246,3 +258,209 @@ def test_enable_v2_versioning_for_scheduled_changes(
+    # of a conflict with another change set that we will create
+    # afterwards and publish immediately).
+    change_request_1 = ChangeRequest.objects.create(
+        title="CR 1",
+        environment=environment_v2_versioning,
+        user=staff_user,
+    )
+    change_set_to_publish = VersionChangeSet.objects.create(
+        feature=feature,
+        change_request=change_request_1,
+        live_from=live_from,
+        feature_states_to_update=json.dumps(
+            [
+                {
+                    "feature": feature.id,
+                    "enabled": True,
+                    "feature_segment": None,
+                    "feature_state_value": {"type": "unicode", "string_value": "foo"},
+                }
+            ]
+        ),
+    )
+    with mock.patch(
+        "features.versioning.tasks.publish_version_change_set"
+    ) as mock_publish:
+        change_request_1.commit(staff_user)
+        task_kwargs = {
+            "version_change_set_id": change_set_to_publish.id,
+            "user_id": staff_user.id,
+            "is_scheduled": True,
+        }
+        mock_publish.delay.assert_called_once_with(
+            kwargs=task_kwargs,
+            delay_until=live_from,
+        )
+
+    # Now, we'll create another change request and change set that will
+    # conflict with the first one and publish it immediately.
+    conflict_change_request = ChangeRequest.objects.create(
+        title="Conflict CR",
+        environment=environment_v2_versioning,
+        user=staff_user,
+    )
+    conflict_feature_value = "bar"
+    conflict_feature_enabled = False
+    VersionChangeSet.objects.create(
+        feature=feature,
+        change_request=conflict_change_request,
+        feature_states_to_update=json.dumps(
+            [
+                {
+                    "feature": feature.id,
+                    "enabled": conflict_feature_enabled,
+                    "feature_segment": None,
+                    "feature_state_value": {
+                        "type": "unicode",
+                        "string_value": conflict_feature_value,
+                    },
+                }
+            ]
+        ),
+    )
+    conflict_change_request.commit(staff_user)
+
+    # When
+    # We simulate the task being called by the task processor at the correct time,
+    # as per the mock call that we asserted above.
+    with freezegun.freeze_time(live_from):
+        publish_version_change_set(**task_kwargs)
+
+    # Then
+    # an alert was sent to the change request owner
+    assert len(mailoutbox) == 1
+    assert mailoutbox[0].subject == change_request_1.email_subject
+    assert mailoutbox[0].to == [staff_user.email]
+    assert mailoutbox[0].body == render_to_string(
+        "versioning/scheduled_change_failed_conflict_email.txt",
+        context={
+            "change_request": change_request_1,
+            "user": staff_user,
+            "feature": feature,
+        },
+    )
```

Comment:
Ah I wish I knew about using the `render_to_string` function for the email testing code I had written for the API usage.